base.sql

```diff
@@ -0,0 +1,130 @@
+  tax integer not null check(tax >= 0)
+);
+
+--house - таблица информации о домах. Цена за аренду определяется следующим образом: арендодатель назначает постоянную цену, которая будет показана сервисом, при этом может дополнительно задать промежутки, в которых стоимость меняется и указывает её, дополнительные цены хранятся в таблице price_for_week 
+--Так же пользователь может указать цену за клининг   
+DROP TABLE IF EXISTS house CASCADE;
+create table house (
+  id SERIAL primary key,
+  country_name varchar(255)not null references country(name_of_country),
+  adress varchar(255) not null,
+  gps lseg not null,
+  description_ text not null,
+  room_number integer not null,
+  beds_number integer not null,
+  max_people integer not null,
+  default_price integer not null,
+  cleaning_price integer,
+  check (room_number > 0 and beds_number > 0 and max_people > 0),
+  check (default_price > 0 and cleaning_price > 0)
+);
+
+--price_for_week - таблица с указанием цены за неделю на определенные даты, устанавливается арендодателем
+DROP TABLE IF EXISTS price_for_week CASCADE;
+create table price_for_week(
+  house_id integer not null references house(id),
+  price integer not null,
+  date_from date not null,
+  date_to date not null, 
+  check (date_from <= date_to and price >= 0), -- проверяем правильность дат
+  unique (house_id, price, date_from, date_to) -- таким образом исключаем ситуации, когда у одного дома разная цена за один и тот же промежуток времени
+);
+
+-- application_rent - информация о датах аренды и сатусе заявки. По дефолту заявка под рассмотрением.
+create type app_status as enum ('accepted', 'declined', 'under consideration');
+DROP TABLE IF EXISTS application_rent CASCADE;
+create table application_rent(
+  id SERIAL primary key,
+  date_residence_start date not null,
+  date_residence_end date not null,
+  house_id int references house(id),
+  descr_of_aplication varchar(255),
+  check (date_residence_start <= date_residence_end), -- проверяем правильность дат
+  status app_status not null default 'under consideration'
+);
+
+--comfort - справочник возможных удобств 
+DROP TABLE IF EXISTS comfort CASCADE;
+create table comfort(
+  comfort_name varchar(255) primary key
+);
+
+--comfort_with_house - таблица о возможных удобств для дома 
+DROP TABLE IF EXISTS comfort_with_house CASCADE;
+create table comfort_with_house (
+    id_house integer not null references house(id),
+    comfort varchar(255) not null references comfort(comfort_name),
+    primary key (id_house, comfort)
+); 
+
+-- reviews_for_house - таблица отзыва на жилье от арендатора. estimation, location_convenience, cleanliness, friendlines, addition_value - отзывы от 1 до 5  
+DROP TABLE IF EXISTS reviews_for_house CASCADE;
+create table reviews_for_house(
+  user_id integer references user_(id),
+  house_id integer references house(id),
+  application_id integer references application_rent(id),
+  estimation smallint not null,
```

Comment:
estimation -- это оценка в смысле "прогноз".
"how can you estimate your database knowledge level?" "excellent!"
"how did they score your test on databases?" "well, it might've been scored higher..."

Reply:
Заменили на evaluation.